gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py

```diff
@@ -60,63 +63,145 @@ def __init__(self, info_keys_to_move_to_calls_regex, copy_quality_to_calls,
+      merged_non_variant = copy.deepcopy(overlapping_variants[0])
+      merged_non_variant.reference_bases = None
+      merged_non_variant.start = start
+      merged_non_variant.end = end
+
+      for non_variant in overlapping_variants[1:]:
+        merged_non_variant.names.extend(non_variant.names)
+        merged_non_variant.filters.extend(non_variant.filters)
+        merged_non_variant.quality = min(merged_non_variant.quality,
+                                         non_variant.quality)
+        merged_non_variant.calls.extend(non_variant.calls)
+
+      merged_non_variant.names = sorted(set(merged_non_variant.names))
+      merged_non_variant.filters = sorted(set(merged_non_variant.filters))
+      merged_non_variant.calls = sorted(merged_non_variant.calls)
+
+      merged_non_variants.append(merged_non_variant)
+
+    return merged_non_variants
+
+  def _merge_variants(self, grouped_variants):
+    merged_variants = []
+    for merge_key, variants in grouped_variants.iteritems():
+      merged_variants.extend(
+          self._move_to_calls.get_merged_variants(variants, merge_key))
+    return merged_variants
+
+  def _update_splits(self, splits, v):
+    start = v.start
+    end = v.end
+    # Increase split size to also combine adjacent splits
+    overlapping_splits = splits.search(v.start - 1, v.end + 1)
+    if overlapping_splits:
+      for split in overlapping_splits:
+        splits.remove(split)
+        start = min(start, split.begin)
+        end = max(end, split.end)
+    splits.addi(start, end, True)
+
+  def _split_non_variants(self, non_variant_tree, splits):
+    for split in splits:
+      overlapping_non_variants = non_variant_tree.search(split.begin, split.end)
+      for onv_interval in overlapping_non_variants:
+        non_variant_tree.remove(onv_interval)
+        onv = onv_interval.data
+        if split.begin <= onv.start and split.end >= onv.end:
+          continue
+        elif split.begin <= onv.start:
+          nv = copy.deepcopy(onv)
+          nv.start = split.end
+          non_variant_tree.addi(nv.start, nv.end, nv)
+        elif split.end >= onv.end:
+          nv = copy.deepcopy(onv)
+          nv.end = split.begin
+          non_variant_tree.addi(nv.start, nv.end, nv)
+        else:
+          left_nv = copy.deepcopy(onv)
+          right_nv = copy.deepcopy(onv)
+          left_nv.end = split.begin
+          right_nv.start = split.end
+          non_variant_tree.addi(left_nv.start, left_nv.end, left_nv)
+          non_variant_tree.addi(right_nv.start, right_nv.end, right_nv)
+
+    return [nv.data for nv in non_variant_tree]
+
+  def _align_with_window(self, v, key):
+    if not key:
+      return
+    start = self._get_window_start(key)
+    end = start + self._window_size
+    if v.start < start:
+      v.start = start
+    if v.end > end:
+      v.end = end
+
+  def _is_non_variant(self, v):
+    if not v.alternate_bases:
+      return True
+    for alternate in NON_VARIANT_ALTERNATES:
```

Comment:
nit: you can write
return any(v.alternate_bases == [a] for a in NON_VARIANT_ALTERNATES)

Reply:
Good call! Done.